qemu/tests/nvdimm.py

```diff
@@ -0,0 +1,138 @@
+
+    def format_nvdimm(self):
+        """
+        Format nvdimm device in guest
+        """
+        format_cmd = self.params["format_command"]
+        status, output = self.session.cmd_status_output(format_cmd)
+        if status != 0:
+            raise error.TestFail("Format command failed: %s" % output)
+
+    def mount_nvdimm(self, format_device="yes"):
+        """
+        Mount nvdimm device in guest
+
+        :param format_device: A string to specify if format the device or not
+        """
+        if format_device == "yes":
+            self.format_nvdimm()
+        mount_cmd = self.params["mount_command"]
+        status, output = self.session.cmd_status_output(mount_cmd)
+        if status != 0:
+            raise error.TestFail("Mount nvdimm device failed: %s" % output)
+
+    def umount_nvdimm(self):
+        """
+        Umount nvdimm device in guest.
+        """
+        umount_cmd = "umount %s" % self.params["pmem"]
+        status, output = self.session.cmd_status_output(umount_cmd)
+        if status != 0:
+            raise error.TestFail("Umount nvdimm device failed: %s" % output)
+
+
+@error.context_aware
+def run(test, params, env):
+    """
+    Run nvdimm cases:
+    1) Boot guest with nvdimm device backed by a host file
+    2) Login to the guest
+    3) Check nvdimm in monitor and guest
+    4) Format nvdimm device in guest
+    5) Mount nvdimm device in guest
+    6) Create a file in the mount point in guest
+    7) Check the md5 value of the file
+    8) Umount the nvdimm device and check calltrace in guest
+    9) Reboot the guest
+    10) Remount nvdimm device in guest
+    11) Check if the md5 value of the nv_file changes
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    nvdimm_test = NvdimmTest(params, env)
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    try:
+        error.context("Login to the guest", logging.info)
+        login_timeout = int(params.get("login_timeout", 360))
+        nvdimm_test.session = vm.wait_for_login(timeout=login_timeout)
+        error.context("Verify nvdimm in monitor and guest", logging.info)
+        nvdimm_test.verify_nvdimm(vm)
+        cmd_timeout = int(params.get("cmd_timeout", 360))
+        error.context("Format and mount nvdimm in guest", logging.info)
+        nvdimm_test.mount_nvdimm()
+        nv_file = params.get("nv_file", "/mnt/nv")
+        error.context("Create a file in nvdimm mount dir in guest, and get "
+                      "original md5 of the file", logging.info)
+        dd_cmd = "dd if=/dev/urandom of=%s bs=1K count=200" % nv_file
+        status, output = nvdimm_test.session.cmd_status_output(dd_cmd,
+                                                               timeout=cmd_timeout)
+        if status != 0:
+            raise error.TestFail("%s failed: %s" % (dd_cmd, output))
+
+        md5_cmd = "md5sum %s" % nv_file
```

Comment:
def run_guest_cmd(session, cmd, check_status=True):
      status, output = session.cmd_status_output(cmd)
      if check_status and status != 0:
           raise excpetions.TestError("Execute command '%s' failed, output: %s")
      return output.strip()

def  md5_hash(session, file):
       cmd = "md5sum %s" % file
       return run_guest_cmd(session, cmd)
rewrite  mount/ format/umount like md5_has  do it